File: lib/models/transaction_model.dart
import 'package:isar/isar.dart';

part 'transaction_model.g.dart';

@Collection()
class TransactionModel {

  Id id = Isar.autoIncrement;

  late DateTime timestamp;
  late double amount;
  late String title;
  late String message;

  @Index(type: IndexType.value)
  late String category;

  @Index(type: IndexType.value)
  late String type;

  TransactionModel({
    required this.timestamp,
    required this.amount,
    required this.title,
    required this.message,
    required this.category,
    required this.type
  });

}

File: src/models/transaction_model.js
import { v4 as uuidv4 } from 'react-native-uuid';

class TransactionModel {
  constructor({ timestamp, amount, title, message, category, type }) {
    this.id = uuidv4();
    this.timestamp = timestamp;
    this.amount = amount;
    this.title = title;
    this.message = message;
    this.category = category;
    this.type = type;
  }
}

export default TransactionModel;